packages/select/src/index.mts

```diff
@@ -49,7 +53,11 @@ export default createPrompt<string, SelectConfig>((config, done) => {
       const offset = isUpKey(key) ? -1 : 1;
       let selectedOption;
 
-      while (!selectedOption || selectedOption.disabled) {
+      while (
+        !selectedOption ||
+        selectedOption.disabled ||
+        selectedOption.type === 'separator'
```

Comment:
This logic is reused ~3 times. Let's extract to a utility `isSelectableChoice()`